### DISCUSSION TITLE
clover sees nvme but won't boot

### DISCUSSION THREAD

**Discussion by pterotronic** [CloverHackyColor/CloverBootloader#672] [Q&A]
i put clover on usb. then copied NvmExpressDxe.efi file to Bios and uefi directories in the drivers sub-directory. Hit function key at boot to switch to bios to choose a boot disk. Select the usb w clover. Clover menu comes up. the new nvme on pci-e slot shows up, so clover sees it. But when you select it, you just get black screen and it freezes there. what's wrong? the nvme is visible both in explorer and disk manager. I cloned the os from C: to nvme. I also tried copying NvmExpressDxe.efi to the nvme drive in the windows, system32, and system directory. No change.

**Comment by Morbius01** [comment]
Booting from Clover on a cloned OS does not always work. In fact, it has never worked for me... 

Since you have created a Clover key which seems to work, I would suggest that you try to re-install Windows **_from the Clover menu_**.

If you need further help, could you give more details, such as:
- the mainboard you use
- the NVMe SSD you use
- is it a UEFI or a legacy system ?
- which version of Windows do you wish to use ?

**Comment by alan-rpi** [comment]
>  the nvme is visible both in explorer and disk manager.
When Windows is running it has a driver for the SSD and when you are running Clover off the USB then it has the NVMe driver on the USB, but when the PC firmware boot it does not have a driver and old bios cannot find it.

You need to keep an EFI partition on the HDD as the PC bios boots this. Some people seem to replace the Windows boot file bootX64.efi **on the HDD** with a copy of cloverx64.efi and/or replacing the bootmgfw.efi or bootmgr.efi (need to check which one) with cloverx64.efi. Then the PC BIOS runs Clover.

I also cloned the os from C: to nvme but kept a copy of Windows on the HDD and when I ran bcdboot to create the EFI contents ended up with a Windows Boot Manager on the HDD. On boot this listed both the HDD Windows and the SSD Windows but the SSD one does not work. 

I then modified the Windows Boot Manager configuration and added a Clover entry pointing to the \EFI\CLOVER\CLOVERX64.efi as the first entry. Now the PC boots to the HDD and Clover runs. I get a Clover icon for the HDD EFI (that then runs the Windows Boot Manager) and an icon to the SSD Windows that takes me to the Login screen.

See my post here https://github.com/CloverHackyColor/CloverBootloader/discussions/674 

Update: If you have a bootable system off the HDD then by running EasyEUFI you can add Clover easily to the Windows Boot Manager. I think you will find there are no existing entries other than Windows itself. The alternative is to open the command line windows as Administrator, and run the Windows BCDEdit command. Running it without any parameters will display your current Boot entries. Further parameters allow you to add Clover and make it the first item. See https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/bcdedit-command-line-options?view=windows-11    I've not used it so search the web for guidance. 

I mention in my AskWoody post that I use Macrium Reflect Recovery. MRR is built on Windows PE so you should be able to use any Windows PE or Windows Recovery USB stick to access a non bootable system and then drop down to the Command line (under Advanced options I think) to run diskpart.
